### PR TITLE
codemod: Fix build and enable running on Braid itself

### PR DESCRIPTION
**Pull request by michaeltaranto**
The codemod appears to have been broken through the course of converting to a monorepo, and once fixed we discovered that a downstream dependency is broken (`yoga-layout-prebuilt`) which appears unlikely to be fixed.

Worked with @mrm007 to internalise this dependency from the codemod build, so that we could patch it to unblock features coming through.

Also added support for running codemod over Braid repo itself, rather than just consumer repos.

### CODE REVIEW

File: packages/codemod/src/codemod.ts [hunk 1]
```diff
@@ -85,9 +85,9 @@ export function babelRecast({
 
     return {
       // @ts-expect-error
-      warnings: metadata ? metadata.warnings : [],
+      warnings: metadata && metadata.warnings ? metadata.warnings : [],
```
Comment (mrm007) [inline packages/codemod/src/codemod.ts:88]:
```suggestion
      warnings: metadata?.warnings ?? [],
```

Gold plating (🏅) is my passion

File: packages/codemod/src/codemod.ts [hunk 2]
```diff
@@ -85,9 +85,9 @@ export function babelRecast({
...
       // @ts-expect-error
-      hasChanged: metadata ? metadata.hasChanged : false,
+      hasChanged: metadata && metadata.hasChanged ? metadata.hasChanged : false,
```
Comment (mrm007) [inline packages/codemod/src/codemod.ts:90]:
```suggestion
      hasChanged: metadata?.hasChanged ?? false,
```

File: packages/codemod/src/plugin-deprecate/isBraidImport.test.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+describe('isBraidImport', () => {
+  describe('valid cases', () => {
+    [
+      'braid-design-system',
+      'braid-design-system/css',
+      'braid-src/lib',
+      'braid-src/lib/components',
+      'braid-src/lib/css',
+    ].forEach((source) =>
+      test(`${source}`, () => expect(isBraidImport(source)).toBeTruthy()),
```
Comment (mrm007) [inline packages/codemod/src/plugin-deprecate/isBraidImport.test.ts:12]: 🏅 Do you prefer this form over `test.each`?

File: packages/codemod/src/plugin-deprecate/isBraidImport.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export const isBraidImport = (importSource: string) =>
+  /(braid-design-system(?:\/css)?|(braid-src\/lib(?:\/(css|components))?))$/.test(
```
Comment (mrm007) [inline packages/codemod/src/plugin-deprecate/isBraidImport.ts:2]: Given that this regular expression will be executed a few hundred times, maybe it's worth extracting it to a constant in the module scope. What do you think?

Comment (mrm007) [reply packages/codemod/src/plugin-deprecate/isBraidImport.ts:2]: 🏅 I'd split it into 2 regular expressions, for readability. Something like

```js
regexForBraid.test(importSource) || regexForInternal.text(importSource)
```

For most cases (consumers), the first branch is the one that will return early.

**Review by mrm007 — APPROVED** [review]
Nice one! A few [gold plating] suggestions marked with 🏅

File: packages/codemod/src/plugin-deprecate/isBraidImport.test.ts (hunk 3 above)
Comment (michaeltaranto) [reply packages/codemod/src/plugin-deprecate/isBraidImport.test.ts:12]: Honestly I just forget its there 😄 . I'll update.

**Review by mrm007 — APPROVED** [review]

**Review by askoufis — APPROVED** [review]

**Pull request merged**